include/xtensor/xgenerator.hpp

```diff
@@ -152,6 +154,9 @@ namespace xt
         template <class OR, class OF>
         rebind_t<OR, OF> build_generator(OF&& func) const;
 
+        template <class O = dynamic_shape_type>
```

Comment:
You can also avoid the definition of `dynamic_shape_type` in the class and use its definition here as the default template parameter....